src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -894,7 +894,11 @@ static FieldRef applyFunction(const FunctionBasePtr & func, const DataTypePtr &
     if (field.isExplicit())
         return applyFunctionForField(func, current_type, field);
 
-    String result_name = "_" + func->getName() + "_" + toString(field.column_idx);
+    WriteBufferFromOwnString buf;
+    writeText("_", buf);
+    writePointerHex(func.get(), buf);
```

Comment:
Sounds insecure.

Reply:
Missing comments.

Reply:
This function is only used in `applyMonotonicFunctionsChainToRange`. So you can just pass the index of function from [array](https://github.com/ClickHouse/ClickHouse/blob/008d02880bdf99fa659d34790d5ed1e9b884b6d8/src/Storages/MergeTree/KeyCondition.cpp#L2543) as a column name.

Reply:
but I think the lenth of the functions should be 1.
elements `intDiv(a, 20)` and `intDiv(a, 50000)` have there own `monotonic functions chain` which length is 1.

Reply:
@alexey-milovidov @CurtizJ PTAL, we need this fix

Reply:
Well, you are right. Maybe using a pointer as a name is not that bad.